torch/_inductor/ir.py

```diff
@@ -4166,14 +4179,21 @@ def is_not_write(arg):
             x.name for x in kernel._schema.arguments if x.kwarg_only
         ]
 
+    def is_legacy_abi_kernel(self):
```

Comment:
Not sure if we really need this. For these two kernels, all optional arguments have real values, so we wouldn't hit the c10::nullopt path. Moreover, we could add another version of interface, e.g. ```aoti_torch__scaled_dot_product_flash_attention_nullopt``` to handle the missing nullopt cases. Relying on this ```is_legacy_abi_kernel``` looks very hacky to me.

Reply:
>  For these two kernels, all optional arguments have real values, so we wouldn't hit the c10::nullopt path.

We would be changing the way the real values get passed in though (regardless of whether we are doing the ShimOptional or pointer approach).

But okay I am happy to do the other less hacky approach once we figure out the issues with getting the valid type above.

Reply:
Even if we add new versions of these two interfaces (actually we only need to care about the flash_attention one and the other is not used in prod), to make sure the newly published snapshots work with old predictor binaries, we still need to keep the branch until the new API is released in predictor, correct?

Reply:
If `repeat_interleave_Tensor` is not in production, let's fix it by changing its shim API. cc @adnanaziz

Reply:
I think you meant to tag @aakhundov :)

Just checked with him, he says it's fine to change